### PR TITLE
auto generate contributors if a contributors list exists but none are…

### PR DESCRIPTION
**Pull request by kendallgassner**
When i first add auto all contributors to a project I have to back track and add the all contributors list to my repo
that is fine but if a list exists in my rc can auto check if it is initialized in my readme and if it isnt can it add it?

<details>
  <summary>📦 Published PR as canary version: <code>under canary scope @auto-canary@9.56.1-canary.1575.19347.0</code></summary>
  <br />
  
  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @auto-canary/bot-list@9.56.1-canary.1575.19347.0
  npm install @auto-canary/auto@9.56.1-canary.1575.19347.0
  npm install @auto-canary/core@9.56.1-canary.1575.19347.0
  npm install @auto-canary/all-contributors@9.56.1-canary.1575.19347.0
  npm install @auto-canary/brew@9.56.1-canary.1575.19347.0
  npm install @auto-canary/chrome@9.56.1-canary.1575.19347.0
  npm install @auto-canary/cocoapods@9.56.1-canary.1575.19347.0
  npm install @auto-canary/conventional-commits@9.56.1-canary.1575.19347.0
  npm install @auto-canary/crates@9.56.1-canary.1575.19347.0
  npm install @auto-canary/docker@9.56.1-canary.1575.19347.0
  npm install @auto-canary/exec@9.56.1-canary.1575.19347.0
  npm install @auto-canary/first-time-contributor@9.56.1-canary.1575.19347.0
  npm install @auto-canary/gem@9.56.1-canary.1575.19347.0
  npm install @auto-canary/gh-pages@9.56.1-canary.1575.19347.0
  npm install @auto-canary/git-tag@9.56.1-canary.1575.19347.0
  npm install @auto-canary/gradle@9.56.1-canary.1575.19347.0
  npm install @auto-canary/jira@9.56.1-canary.1575.19347.0
  npm install @auto-canary/maven@9.56.1-canary.1575.19347.0
  npm install @auto-canary/npm@9.56.1-canary.1575.19347.0
  npm install @auto-canary/omit-commits@9.56.1-canary.1575.19347.0
  npm install @auto-canary/omit-release-notes@9.56.1-canary.1575.19347.0
  npm install @auto-canary/pr-body-labels@9.56.1-canary.1575.19347.0
  npm install @auto-canary/released@9.56.1-canary.1575.19347.0
  npm install @auto-canary/s3@9.56.1-canary.1575.19347.0
  npm install @auto-canary/slack@9.56.1-canary.1575.19347.0
  npm install @auto-canary/twitter@9.56.1-canary.1575.19347.0
  npm install @auto-canary/upload-assets@9.56.1-canary.1575.19347.0
  # or 
  yarn add @auto-canary/bot-list@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/auto@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/core@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/all-contributors@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/brew@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/chrome@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/cocoapods@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/conventional-commits@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/crates@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/docker@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/exec@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/first-time-contributor@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/gem@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/gh-pages@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/git-tag@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/gradle@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/jira@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/maven@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/npm@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/omit-commits@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/omit-release-notes@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/pr-body-labels@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/released@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/s3@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/slack@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/twitter@9.56.1-canary.1575.19347.0
  yarn add @auto-canary/upload-assets@9.56.1-canary.1575.19347.0
  ```
</details>

### CODE REVIEW

File: plugins/all-contributors/src/index.ts [hunk 1]
```diff
@@ -485,31 +509,37 @@ export default class AllContributorsPlugin implements IPlugin {
...
+
+        const displayFile = file ? fs.readFileSync(file, 'utf8') : '';
+
+        const notInitalized = displayFile.indexOf(
+          '<!-- markdownlint-disable -->\n<!-- markdownlint-restore -->'
+        );
+
+        if (notInitalized && file) {
+          await Promise.all(
+            generateContributorReadme(config, undefined)
```
Comment (hipstersmoothie) [inline plugins/all-contributors/src/index.ts:537]: should we pass config.contributors here?

File: plugins/all-contributors/src/index.ts [hunk 2]
```diff
@@ -173,20 +173,44 @@ function getExtraContributors(body?: string) {
...
+        ...config,
+        contributors,
+      },
+      contributors,
+      oldReadMe
+    );
+    fs.writeFileSync(file, newReadMe);
+  })
+} 
+
```
Comment (hipstersmoothie) [inline plugins/all-contributors/src/index.ts:195]:
```suggestion
/** Determine which files need to display contributors and generate contributors */
async function generateContributorReadme(
  config: AllContributorsRc,
  contributors: any
) {
  return Promise.all(
    (config.files || ["README.md"]).map(async (file) => {
      const oldReadMe = fs.readFileSync(file, {
        encoding: "utf-8",
      });
      const newReadMe = await generateReadme(
        {
          contributorsPerLine: 7,
          imageSize: 100,
          ...config,
          contributors,
        },
        contributors,
        oldReadMe
      );
      fs.writeFileSync(file, newReadMe);
    })
  );
}
```

File: plugins/all-contributors/src/index.ts [hunk 3]
```diff
@@ -485,31 +509,37 @@ export default class AllContributorsPlugin implements IPlugin {
...
+        const displayFile = file ? fs.readFileSync(file, 'utf8') : '';
+
+        const notInitalized = displayFile.indexOf(
+          '<!-- markdownlint-disable -->\n<!-- markdownlint-restore -->'
+        );
+
+        if (notInitalized && file) {
+          await Promise.all(
+            generateContributorReadme(config, undefined)
+         );
```
Comment (hipstersmoothie) [inline plugins/all-contributors/src/index.ts:538]:
```suggestion
          await generateContributorReadme(config, config.contributors)
```

Comment (hipstersmoothie) [reply plugins/all-contributors/src/index.ts:538]: And then change the first usage as well

File: plugins/all-contributors/src/index.ts (hunk 1 above)
Comment (kendallgassner) [reply plugins/all-contributors/src/index.ts:537]: doesnt that already happen with  ...config?

**Review by hipstersmoothie — APPROVED** [review]

**Comment by adierkens** [comment]
:rocket: PR was released in [`v9.57.0`](https://github.com/intuit/auto/releases/tag/v9.57.0) :rocket:

**Pull request merged**